### ISSUE TITLE
`patch` utility not installed

### ISSUE THREAD

**Issue by infotroph** [rocker-org/rocker-versioned2#605]
### Container image name

rocker/rstudio:devel

### Container image digest

rocker/rstudio@sha256:043f6aaa7ad27050fb4395c3a14c8e6bb8a334ce3645bd412812b978c0b3324c

### What operating system are you seeing the problem on?

Linux

### System information

First seen on GitHub Actions using ubuntu-latest, reproduced locally on MacOS 12.6.3 (ARM) with Docker 20.10.22, build 3a2c30b

### Bug description

the Rocker versioned stack doesn't appear to have the `patch` utility installed, leading to compilation failures in packages that use it unconditionally in their build scripts, e.g. hdf5r 1.3.8

The easy workaround in my case is `apt-get install patch`, but I was surprised to need it on a system that has the rest of the GCC toolchain. Was patch left off the base image intentionally or inadvertently? Would you consider adding it?

### How to reproduce this bug?

```sh
Minimal reprex: 
  `docker run --rm -it rocker/rstudio:devel which patch`
  Expected output: `/usr/bin/patch`
  Actual output: [none]

Reprex with a more compelling use case:


chrisb@ulmus ~ % docker run --rm -it rocker/rstudio:devel /bin/bash
root@77700e44609e:/# apt-get update && apt-get install -y libhdf5-dev
[...]
root@77700e44609e:/# R -e 'install.packages("hdf5r")'
[...]
* installing *source* package ‘hdf5r’ ...
[...]
using C compiler: ‘gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0’
cp 1_10_3/*.c 1_10_3/*.h .
if [ "1.10.6" != "" ]; then patch -p2 < patch_1.10.6; fi
/bin/bash: line 1: patch: command not found
make: *** [Makevars:28: copy_auto_files] Error 127
ERROR: compilation failed for package ‘hdf5r’
* removing ‘/usr/local/lib/R/site-library/hdf5r’

The downloaded source packages are in
	‘/tmp/RtmpRCNf5S/downloaded_packages’
Warning message:
In install.packages("hdf5r") :
  installation of package ‘hdf5r’ had non-zero exit status
```

**Comment by eitsupi** [comment]
@cboettig Any thoughts?

**Comment by cboettig** [comment]
Thanks @infotroph for your question.  We install necessary system dependencies R, users need to add additional libraries to install additional packages, as you have noted above.  I'm not quite clear in what sense you mean by `patch` being considered a 'core part of the gcc' toolchain.  Can you be more precise?  We are following the Ubuntu (server/docker image, not desktop) distribution definitions here.

**Comment by eddelbuettel** [comment]
Second this. There is `build-essential` and it is installed; `patch` is an auxiliary a user should install.

**Comment by benz0li** [comment]
> There is `build-essential` and it is installed;

@eddelbuettel Are you sure? Because [build-essential](https://packages.ubuntu.com/jammy/build-essential) depends on `dpkg-dev` depends on `patch`.

**Comment by infotroph** [comment]
Thanks. I do see that the DESCRIPTION for {hdf5r} includes "GNU make" in the SystemRequirements field -- are you saying they should additionally list "patch" as a separate requirement? Or does `build-essential` not map to "GNU make"?

**Comment by eddelbuettel** [comment]
@benz0li You are right, I was wrong.  But arguably `dpkg-dev` is distro-specific, the "build essential" tools are really the compiler, binutils, and a build tool (`make`).

**Comment by eddelbuettel** [comment]
@infotroph Try `apt-cache show build-essential` to see.  Also note that CRAN Policy, at least in spirit, only mentions "unusual" requirements (think: `cmake`, not common at CRAN) or `GNU make` as opposed to a more minimal make.

The bigger picture is that our containers have to strike a balance between what is useful for _most to all_ users and not too many somewhat useful but more peripheral programs.  For you today this is `patch`, these days people more commonly start with `git` which one _can_ think of as a generalization of `patch` and `diff` (and then some).

So in short: for this one package you are after, please just install `patch` with `apt`.

**Comment by infotroph** [comment]
For any given downstream use, I know it's always ultimately on me to install what I need. Expected, reasonable, no complaints.

That said, I'm having a hard time squaring @cboettig's "we follow the Ubuntu distribution definitions" (Which I'm guessing means the `Build-Essential: yes` tag) with @eddelbuettel's "dpkg-dev is distro-specific." Whose definition of "essential"/"necessary" should I be looking at before asking here?

**Comment by eddelbuettel** [comment]
"Both". 

We try to follow what the distro does (and I am also a Debian maintainer) yet strike a balance of keeping the containers "at a reasonable size and not too large".  (And what I said [here](https://github.com/rocker-org/rocker-versioned2/issues/605?notification_referrer_id=NT_kwDOAApFYbE1NzY5NjU5NDg2OjY3MzEyMQ#issuecomment-1455153570) I corrected [a post later](https://github.com/rocker-org/rocker-versioned2/issues/605?notification_referrer_id=NT_kwDOAApFYbE1NzY5NjU5NDg2OjY3MzEyMQ#issuecomment-1455163143) as @benz0li kindly pointed out that I was off track.)

We simply _cannot_ satisfy _all requests_ by _all_ users at _all_ times. Asymptotically we'd end up installing all of GitHub.  So sometimes we have to make editorial choices.  We try to explain this one to you in this thread.  If you peruse the history of issues in our repos (which are a little scattered around the github org, sorry) you will see quite a few similar prior discussions.  We often have to say no.

That said, sometimes we get it wrong, omit something, or forgot to include something.  `patch` is borderline, but we all at the Rocker team seem to feel it is on the other side of the border.

**Comment by infotroph** [comment]
So in other words this is a place you *don't* follow what the distro does. That's your prerogative and I _am not asking you to change it_, but since y'all started the thread by claiming you do follow it, I do not like being yelled at for being confused.
